client/src/components/newSource/index.tsx

```diff
@@ -226,7 +195,28 @@ const NewSourceView = (): React.ReactElement => {
         } catch (e) {
           reject(new Error(`could not parse ${fhirBundleFile!.name} as JSON`));
         }
-        bundleJson.entry.map(uploadFhirResource);
+        // Check that we only have StructureDefinitions, ConceptMaps,
+        // or CodeSystems in the bundle
+        const authorizedTypes = [
+          'StructureDefinition',
+          'ConceptMap',
+          'CodeSystem'
+        ];
+        const resourceTypes = bundleJson.entry.map(
+          (entry: any) => entry.resourceType
+        );
+        if (
+          resourceTypes.some((type: string) => !authorizedTypes.includes(type))
+        )
+          reject(
+            new Error(
+              `Source created but failed while trying to upload a bundle with 
+              resource which type is not
+              StructureDefinition, ConceptMap, nor CodeSystem`
+            )
+          );
+        // Upload the bundle
+        await axios.post(`${FHIR_API_URL}/upload-bundle`, bundleJson);
```

Comment:
what's the maximum size that the API accepts ? maybe we should mention it "your bundle cannot exceed 1mb" or something)

Reply:
No idea, not sure how this is handle by flask but I'll have a look